src/classes/frDonation.cls

```diff
@@ -53,84 +55,126 @@ public class frDonation extends frModel implements frSyncable {
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
+        String supporterFunraiseId = String.valueOf(request.get('donorId')); 
+        String supporterObjName = frUtil.hasPersonAccountsEnabled() ? 'Account' : 'Contact';
+        String subsciptionObjName = frUtil.hasNPCobjects() ? 'GiftCommitment' : 'Subscription__c';
+        
+        //connect supporter
         if (String.isNotBlank(supporterFunraiseId)) {
-            List<Contact> contacts = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
-            if (contacts.size() > 0) {
-                Contact supporterContact = contacts.get(0);
-                o.fr_Donor__c = supporterContact.Id;
-                if(frUtil.hasNPSP()){
-                    o.put('npsp__Primary_Contact__c', supporterContact.Id);
-                    o.AccountId = supporterContact.AccountId;
-                }
+            List<SObject> supporters = Database.query('SELECT Id FROM ' + supporterObjName + ' WHERE fr_ID__c = :supporterFunraiseId');
+            if (supporters.size() > 0) {
+                SObject supporter = supporters.get(0);
+                o.put('DonorId', supporter.get('Id'));
             } else {
                 if(createLogRecord) frUtil.logRelationshipError(getFrType(), funraiseId, 
                                             frUtil.Entity.SUPPORTER, supporterFunraiseId);
             }
         }
-        
+        //connect campaign
         Boolean campaignMappingDisabled = request.containsKey(META_CAMPAIGN_MAPPING_KEY) ?
-            Boolean.valueOf(request.get(META_CAMPAIGN_MAPPING_KEY)): true;
+                Boolean.valueOf(request.get(META_CAMPAIGN_MAPPING_KEY)): true;
         if(!campaignMappingDisabled) {
             String campaignId = String.valueOf(request.get('campaignGoalId'));
             if (String.isNotBlank(campaignId)) {
                 List<Campaign> campaigns = [SELECT Id FROM Campaign WHERE fr_ID__c = :campaignId];
                 if (campaigns.size() > 0) {
-                    o.CampaignId = campaigns[0].Id;
+                    o.put('CampaignId', campaigns[0].Id);
                 } else {
                     frUtil.logRelationshipError(getFrType(), funraiseId, 
                                                 frUtil.Entity.CAMPAIGN, campaignId);
                 }
             }
         }
-        
+        //connect subscription
         String subscriptionId = String.valueOf(request.get('subscriptionId'));
         if (String.isNotBlank(subscriptionId)) {
-            List<Subscription__c> subscriptions = [SELECT Id FROM Subscription__c WHERE fr_ID__c = :subscriptionId];
+            List<SObject> subscriptions = Database.query('SELECT Id FROM ' + subsciptionObjName + ' WHERE fr_ID__c = :subscriptionId');
             if (subscriptions.size() > 0) {
-                o.Subscription__c = subscriptions.get(0).Id;
+                o.put('GiftCommitmentId', subscriptions.get(0).get('Id'));
```

Comment:
It looks like this removed the the non-NPC logic where we should be populating the Subscription__c field on the opportunity when we aren't in an NPC environment

Reply:
Added in new update!